src/rpp/rpp/schedulers/trampoline_scheduler.hpp

```diff
@@ -0,0 +1,92 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2022 - present.
+//                            TC Wang 2022 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/schedulers/fwd.hpp>                       // own forwarding
+#include <rpp/schedulers/details/worker.hpp>            // worker
+#include <rpp/subscriptions/composite_subscription.hpp> // lifetime
+#include <rpp/schedulers/details/queue_worker_state.hpp>// state
+
+#include <concepts>
+#include <chrono>
+#include <functional>
+#include <thread>
+
+thread_local bool queue_in_use = false;
+thread_local rpp::schedulers::details::queue_worker_state queue{};
+
+namespace rpp::schedulers
+{
+
+template <typename F>
+struct scope_guard {
+    explicit scope_guard(F&& f) : m_f(std::forward<F>(f)) {}
```

Comment:
You need move instead of forward there. `&&` is forwarding **ONLY** when you have template argument for this one FOR function.
```cpp
template <typename F>
struct scope_guard {
    template<typename U>
    explicit scope_guard(U&& f) : m_f(std::forward<U>(f)) {}
```
^^^^^ is forwarding reference due to it is deduced FROM argument of function (constrctor). But in your case you class specialized with template and then you accept it by **rvalue-reference**

Reply:
You're right. Thanks for catching this :)